addon/components/sl-select.js

```diff
@@ -141,58 +209,83 @@ export default Ember.Component.extend( InputBased, TooltipEnabled, {
                     }
 
                     return output;
-                },
-
-                formatSelection: ( item ) => {
-                    if ( !item ) {
-                        return null;
-                    }
-
-                    const typeOfItem = Ember.typeOf( item );
-
-                    if (
-                        'object' === typeOfItem ||
-                        'instance' === typeOfItem
-                    ) {
-                        return Ember.get( item, this.get( 'optionLabelPath' ) );
-                    }
+                }
 
-                    return item;
-                },
+                return item;
+            };
+        }
+    ),
 
-                id: ( item ) => {
-                    let value = item;
-                    const typeOfItem = Ember.typeOf( item );
+    /**
+     * Select2 fomartSelection function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2FormatSelection: Ember.computed(
+        function() {
+            return ( item ) => {
+                const typeOfItem = Ember.typeOf( item );
+
+                if (
+                    'object' === typeOfItem ||
+                    'instance' === typeOfItem
+                ) {
+                    return Ember.get( item, this.get( 'optionLabelPath' ) );
+                }
 
-                    if (
-                        'object' === typeOfItem ||
-                        'instance' === typeOfItem
-                    ) {
-                        const optionValuePath = this.get( 'optionValuePath' );
-                        value = Ember.get( item, optionValuePath );
-                    }
+                return item;
+            };
+        }
+    ),
 
-                    return value;
-                },
+    /**
+     * Select2 id function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2Id: Ember.computed(
+        function() {
+            return ( item ) => {
+                const typeOfItem = Ember.typeOf( item );
+
+                if (
+                    'object' === typeOfItem ||
+                    'instance' === typeOfItem
+                ) {
+                    return Ember.get( item, this.get( 'optionValuePath' ) );
+                }
 
-                initSelection: ( element, callback ) => {
-                    const value = element.val();
+                return item;
+            };
+        }
+    ),
 
-                    if ( !value || !value.length ) {
-                        return callback( [] );
-                    }
+    /**
+     * Select2 initSelection function
+     *
+     * @function
+     * @returns {Function}
+     */
+    select2InitSelection: Ember.computed(
+        function() {
+            return ( element, callback ) => {
+                const value = element.val();
 
+                if ( !value || !value.length ) {
+                    callback( [] );
```

Comment:
according to the select2 documentation, this function shouldn't be returning anything, it should just be using the callback. Not that returning would break anything, but it was unnecessary, since the callback doesn't return anything either.